### PR TITLE
EEPROM cleanup

### PR DESCRIPTION
**Pull request by ewz4**
# EEPROM rework

This is the initial implementation with the calls to the Control Task and Restore commented out. When I run my test cases on it and verify it works I'll uncomment. Has new SFRFields relevant to the normal report and also now using time_alive to check if the initial two hours have passed.

### CODE REVIEW

File: src/EEPROMRestore.cpp [hunk 1]
```diff
@@ -1,50 +1,108 @@
 #include "EEPROMRestore.hpp"
+// #include <algorithm> // find
```
Comment (ljgreenhill) [inline src/EEPROMRestore.cpp:2]: ?

Comment (ewz4) [reply src/EEPROMRestore.cpp]: Gone

File: src/sfr.cpp [hunk 2]
```diff
@@ -282,20 +282,18 @@ namespace sfr {
...
+        SFRField<uint32_t> sfr_data_age = SFRField<uint32_t>(0, 0x2808);
+        SFRField<bool> sfr_save_completed = SFRField<bool>(false, 0x2809);
 
-        // @ERIC fix if needed
-        int wait_time_last_write_time = 0; // the millis() value when the last EEPROM wait time write ocurred, should reset to 0 every boot up cycle
-        int sfr_last_write_time = 0;       // the millis() value when the last EEPROM SFR data write ocurred, should reset every boot up cycle
-        uint16_t sfr_address = 7;          // the address of where current SFR data is stored, read from EEPROM and set at the beginning of every boot up cycle
-        // Bytes 0-3 are for the time passed, byte 4 holds the number of reboots, and bytes 5-6 are for the address where values are to be written and stored.
-        // SFR data begins at byte 7 and after. The section of EEPROM bytes where SFR data is stored will change to avoid exceeding the write endurance.
+        uint32_t last_fast_save_time = 0;
```
Comment (ljgreenhill) [inline src/sfr.cpp:296]: do these have to be in the SFR

File: src/sfr.hpp [hunk 3]
```diff
@@ -279,16 +279,18 @@ namespace sfr {
...
-        extern SFRField<bool> storage_full;
-        extern SFRField<uint8_t> boot_counter;
-        extern SFRField<uint32_t> sfr_write_step_time;
-        extern SFRField<uint32_t> sfr_address_age;
-
-        // @ERIC fix if needed
-        extern int wait_time_last_write_time;
-        extern int sfr_last_write_time;
-        extern uint16_t sfr_address;
+        extern SFRField<bool> boot_mode;
```
Comment (ljgreenhill) [inline src/sfr.hpp:282]: please but these in order by type

File: src/SFRField.hpp [hunk 4]
```diff
@@ -38,141 +26,78 @@ class SFRInterface
...
-        else if (sizeof(T) == sizeof(bool))
-            data_type = 1;
-
-#ifdef DEBUG
-        T initial = default_val * resolution; // Since value gets set to initial in reset(), initial should be default_val * resolution instead of default_val
-#endif
-
+        this->resolution = resolution;
+        restore = false;
+        T initial = default_val * resolution;
```
Comment (ljgreenhill) [inline src/SFRField.hpp:99]: is initial value used?

**Review by ljgreenhill — APPROVED** [review]

**Pull request merged**